### ISSUE TITLE
Clarification on order structure/ changes to manual

### ISSUE THREAD

**Issue by wardbradt** [ccxt/ccxt#3478]
- OS: N/A
- Programming Language version: N/A
- CCXT version: 1.14.301
- Exchange: N/A
- Method: `fetch_order`

My question is about the `price` key/ value pair in [orders returned](https://github.com/ccxt/ccxt/wiki/Manual#order-structure) from `fetch_order` after the order has been (partially or fully filled). 

If I place an order to sell `BTC/USD` for 1000 with a volume of 1 and there exists two bids on the exchange's order book, one at a rate of 1001 with a volume of 0.3 and the other at 1010 with a volume of 0.8, from my understanding I would sell 0.8 BTC for 1010 and 0.2 BTC for 1001.

When fetching the order with `fetch_order`, after the order has been filled, would `price` be `1000` or `1008.2` (1010 * 0.8 + 1001 * 0.2). If it is `1000`, is the only way to find the effective price the order executed at to either use the `trades` key/value pair in the order or the `fetchMyTrades()`?

-----
Also, I noticed the [Trades By Order Id](https://github.com/ccxt/ccxt/wiki/Manual#trades-by-order-id) section in the manual. Would this be any different than `fetch_order(id)['trades']`? I could delete it and open a PR if not.

**Comment by kroitor** [comment]
> If I place an order to sell BTC/USD for 1000 with a volume of 1 and there exists two bids on the exchange's order book, one at a rate of 1001 with a volume of 0.3 and the other at 1010 with a volume of 0.8, from my understanding I would sell 0.8 BTC for 1010 and 0.2 BTC for 1001.

That's right )

> When fetching the order with fetch_order, after the order has been filled, would price be 1000 or 1008.2 (1010 * 0.8 + 1001 * 0.2).

In many cases that is the initial price of the order. The filling price (if available) is designated in the `average` field, though this was mostly a temporary solution. The `cost`, however, is usually the actual filling cost of the order, so `cost` of a filled order = `filled` * `actual filling price`. So, in that case there's several ways of detecting the actual price in userland without additional calls to fetchTrades:
- compare the `price` to the ordered price
- compare the ordered `cost` to the actual `cost`

Yet still, this is very far from ideal. The robust solution to this is to use `fetchMyTrades` or `fetchOrderTrades` where available at a cost of one additional call.

Your point is understood, of course, as we were thinking of doing some refactoring to that part of the unified order API. It is obvious that those fields, namely, the `amount`, the `cost`, the `price`, should all be properly separated and documented. In the end we should have something like `initialAmount`, `initialPrice`, `initialCost`, and then `filledAmount`, `remainingAmount`, `actualPrice`, `actualCost`. This is in our priority list.

> Would this be any different than fetch_order(id)['trades']? I could delete it and open a PR if not.

The return of `fetchOrderTrades` method should be unified with `fetchOrder(id)['trades']` of course. However, we can't really delete it, because many exchanges have separate endpoints for these two pieces of info.

Let us know if the above does not answer your questions.

**Comment by wardbradt** [comment]
I am a little confused about your last point,

> The return of `fetchOrderTrades` method should be unified with `fetchOrder(id)['trades']` of course. However, we can't really delete it, because many exchanges have separate endpoints for these two pieces of info.

After a call to `fetchOrder(id)`, does it take an additional call to access the value for `trades`? I am confused because you said 

> The robust solution to this is to use `fetchMyTrades` or `fetchOrderTrades` where available at a cost of one additional call.

as well as

> The return of `fetchOrderTrades` method should be unified with `fetchOrder(id)['trades']`

**Comment by kroitor** [comment]
> After a call to `fetchOrder(id)`, does it take an additional call to access the value for `trades`?

Nope. The general policy with requests in the library is to never issue more than one request, unless there's absolutely no other way to solve it, and so far, we've managed to keep it that way mostly. So, it will never issue more than one request per call. And the `fetchOrderTrades` is not called from `fetchOrder`. The `trades` info in the order is taken from the order response directly, if it is provided by the exchange. If not, `trades` will be undefined and therefore you will have to call `fetchMyTrades` or `fetchOrderTrades`, whatever is available, unless you solve it another way by comparing expected to actual values.

> The return of `fetchOrderTrades` method should be unified with `fetchOrder(id)['trades']`

Yep, when I said this, I actually intended to say: "by design, the return of fetchOrderTrades should be the same as the info contained in fetchOrder(id)['trades'], meaning the format of the data, not the amount of requests sent out". And they are unified, however, there may be some places and exchanges 
 where this is not ideal or missing, because, well, you know, we're under development constantly...
